istioctl/pkg/xds/client.go

```diff
@@ -93,3 +97,49 @@ func DialOptions(opts *clioptions.CentralControlPlaneOptions, kubeClient kube.Ex
 		grpc.WithPerRPCCredentials(supplier),
 	}, err
 }
+
+// ApplyXdsFlagDefaults uses values from the injector config for unset flags needed to contact istiod
+func ApplyXdsFlagDefaults(centralOpts *clioptions.CentralControlPlaneOptions, istioNamespace, revision, meshConfigFile string, kubeClient kube.Client) error {
+	// If the user did not specify any XDS address or Istiod selector, get them from injector config
+	if centralOpts.Xds == "" && centralOpts.XdsPodLabel == "" && centralOpts.XdsPodPort == 0 {
+		var meshConfig *v1alpha1.MeshConfig
+		var err error
+		if meshConfigFile != "" {
+			if meshConfig, err = mesh.ReadMeshConfig(meshConfigFile); err != nil {
+				return err
+			}
+		} else {
+			if meshConfig, err = clioptions.GetMeshConfigFromConfigMap(kubeClient, istioNamespace, revision); err != nil {
```

Comment:
Certs will also need to be loaded at some point ( can be 'system' for external or meshes with real certs )